packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -507,17 +566,25 @@ function WorklistDataGrid(props) {
         onKeyDown={handleKeyDown}
         {...(activeIndex != null && { onMouseUp, onMouseMove, onMouseLeave: onMouseUp })}
       >
-        <ColumnHeader
-          columns={dataGridColumns}
-          headerHeight={columnHeaderHeight}
-          tableHeight={tableHeight}
-          tabStopColumnIndex={focusedRow.current === 0 ? focusedCol.current : undefined}
-          onColumnSelect={handleColumnSelect}
-          onResizeMouseDown={onResizeMouseDown}
-        />
-        <tbody>
-          {buildRows(rows)}
-        </tbody>
+        <ColumnContext.Provider
```

Comment:
You don't really need context for this use case.  To pass the value down one level to two components doesn't really seem to require the usage of context.

Reply:
We chose to use context for a few reasons:

- We can bypass the `Row` and `ColumnHeader` components, which don't need to know anything about pinned/overflow columns, and directly pass the data to `Cell` and `ColumnHeaderCell` (the only components that need to know whether or not they are pinned).
- We originally used separate `pinnedColumns` and `overflowColumns` props which were passed through the `Row` and `ColumnHeader` components. This created extra handling within those components that is largely irrelevant to the purpose of those components.
- Passing `pinnedColumns` and `overflowColumns` props down multiple levels of components requires additional consideration for handling cell selection and column resizing due to shared indexes, but using the existing `dataGridColumns` variable (which encompasses all columns) does not break existing functionality.

While it would be nice to have a simple `isPinned` prop on the `Cell` and `ColumnHeaderCell` components, we also need to pass the offsets that are calculated at the `WorklistDataGrid` level, and adding these props to `Row` and `ColumnHeader` serves only as a passthrough and unnecessarily bloats the props for those components.

Reply:
How will this work properly once we memoize the ColumnHeader and Row objects.  The properties passed to these component will not change and thus not rerender.

Reply:
Row and ColumnHeader are already being bypassed so even now, they have no impact on re-rendering pinned-columns. Any changes to the context provider causes the children components to re-render and React.memo does not affect this behavior.

https://react.dev/reference/react/memo#updating-a-memoized-component-using-a-context